### DISCUSSION TITLE
rpc to update multiple rows

### DISCUSSION THREAD

**Discussion by earonesty** [supabase/supabase#15744] [General]
i wrote this rpc and it doesn't work from supabase.  

-  calling the rpc using PERFORM from the cli works fine
-  calling individual inserts works fine
-  calling `rpc("set_source_urls", id, urls).execute()` .... does not work (interestingly it validates the parameters ...but it doesn't actually do the work)

```plpgsql
CREATE OR REPLACE FUNCTION set_source_urls(id text, urls text[])
RETURNS void AS
$$
DECLARE
    url_item text;
BEGIN
    DELETE FROM source_urls WHERE source_id = id;
    
    FOREACH url_item IN ARRAY urls
    LOOP
        INSERT INTO source_urls (source_id, url) VALUES (id, url_item);
    END LOOP;
END;
$$
LANGUAGE plpgsql;
```

**Comment by GaryAustin1** [comment]
Are the inserts from the API or the CLI?  When you run from the API you normally have to  meet RLS on the tables which is different if you are the postgres user.

**Reply by earonesty** [reply]
im using the api

**Comment by feavel1** [comment]
Same issue. Have you found a way to insert multiple rows?

**Comment by steve-chavez** [comment]
> calling rpc("set_source_urls", id, urls).execute() .... does not work

Needs elaboration, was it a permission error? Updating multiple rows through RPC should work normally.
